c7n/resources/s3.py

```diff
@@ -442,21 +449,27 @@ def process(self, buckets):
             results = []
             futures = []
             for b in buckets:
+                region = client.get_bucket_location(
```

Comment:
extraneous api call, also location is not directly mappable to regions in all cases.